src/hevm/src/EVM/Expr.hs

```diff
@@ -221,6 +223,12 @@ readBytes (Prelude.min 32 -> n) idx buf
 
 -- | Reads the word starting at idx from the given buf
 readWord :: Expr EWord -> Expr Buf -> Expr EWord
+readWord idx (WriteWord idx' val buf)
+  | idx == idx' = val
+  | otherwise = readWord idx buf
```

Comment:
maybe we should check if `idx < idx' || idx > idx' + 31` in order to entirely skip this write? If addresses are not word-aligned `idx` could be in the middle of the word being written